libraries/BufferedBleSpp/nanoFramework/BufferedBleSpp.cs

```diff
@@ -0,0 +1,275 @@
+        {
+            GattWriteRequest request = WriteRequestEventArgs.GetRequest();
+
+            byte[] data = new byte[request.Value.Length];
+
+            DataReader rdr = DataReader.FromBuffer(request.Value);
+            rdr.ReadBytes(data);
+
+            #region *// Init the process
+            rxByteAmount = BitConverter.ToInt32(data, 0);
+            rxByteCount = 0;
+            rxByteArray = new byte[0];
+            txAmount = -1;
+            txByteArrayReadCount = 0;
+            txByteArray.Clear(); 
+            #endregion
+
+        }
+
+        private void RxBufferCharacteristic_WriteRequested(GattLocalCharacteristic sender, GattWriteRequestedEventArgs WriteRequestEventArgs)
+        {
+            GattWriteRequest request = WriteRequestEventArgs.GetRequest();
+
+            byte[] data = new byte[request.Value.Length];
+
+            DataReader rdr = DataReader.FromBuffer(request.Value);
+            rdr.ReadBytes(data);
+
+            rxByteArray = Combine(rxByteArray, data);
+            rxByteCount += data.Length;
+
+            if (rxByteCount == rxByteAmount)
+            {
+                // Request received, pass data to application
+                var response = ReceivedData?.Invoke(this, rxByteArray);
+
+                // Return data from application
+                txAmount = response.Length;
+
+
+                #region *// Build list with chunks of bytes buffers from the source array
+                byte[] inBuffer = new byte[BUFFER_SIZE];
+                byte[] outBuffer;
+                using (Stream input = new MemoryStream(response))
+                {
+                    while (input.Position < input.Length)
+                    {
+                        int remaining = BUFFER_SIZE, bytesRead;
+                        while (remaining > 0 && (bytesRead = input.Read(inBuffer, 0,
+                                Math.Min(remaining, BUFFER_SIZE))) > 0)
+                        {
+                            outBuffer = new byte[bytesRead];
+                            Array.Copy(inBuffer, 0, outBuffer, 0, bytesRead);
+                            txByteArray.Add(new Buffer(outBuffer));
+                            remaining -= bytesRead;
+                        }
+                    }
+                }
+                #endregion
+            }
+
+            if (request.Option == GattWriteOption.WriteWithResponse)
+            {
+                request.Respond();
+            }
+
+            // Helper
+            byte[] Combine(byte[] first, byte[] second)
+            {
+                byte[] bytes = new byte[first.Length + second.Length];
+                Array.Copy(first, 0, bytes, 0, first.Length);
+                Array.Copy(second, 0, bytes, first.Length, second.Length);
+                return bytes;
+            }
+
+        }
+
+        
+        private void _txCharacteristic_AmountReadRequested(GattLocalCharacteristic sender, GattReadRequestedEventArgs ReadRequestEventArgs)

```

Comment:
you should follow the naming convention for those 2 ones So TxCharacteristicAmountRequested